components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authzservermetadata/AuthzServerMetadataEndpoint.java

```diff
@@ -0,0 +1,109 @@
+import javax.ws.rs.core.Response;
+
+/**
+ * Rest implementation of OAuth 2 Authorization Server Metadata endpoint.
+ */
+@Path("/{issuer}/.well-known/oauth-authorization-server")
+public class AuthzServerMetadataEndpoint {
+
+    private static final Log log = LogFactory.getLog(AuthzServerMetadataEndpoint.class);
+    private static final String AUTHZ_SERVER_METADATA_ENDPOINT_PATH_COMPONENT_VALUE_TOKEN = "token";
+
+    @GET
+    @Produces("application/json")
+    public Response getAuthzServerMetadata(
+            @PathParam("issuer") String authzServerMetadataEpPathComponent, @Context HttpServletRequest request) {
+
+        String tenantDomain = null;
+        Object tenantObj = IdentityUtil.threadLocalProperties.get().get(OAuthConstants.TENANT_NAME_FROM_CONTEXT);
+        if (tenantObj != null) {
+            tenantDomain = (String) tenantObj;
+        }
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+        if (isValidIssuer(authzServerMetadataEpPathComponent)) {
+            return this.getResponse(request, tenantDomain);
+        } else {
+            Response.ResponseBuilder errorResponse = Response.status(HttpServletResponse.SC_BAD_REQUEST);
+            if (log.isDebugEnabled()) {
+                log.debug("The oauth authorization server metadata path component is " + authzServerMetadataEpPathComponent +
+                        " . The expected oauth authorization server metadata path component is either '"
+                        + AUTHZ_SERVER_METADATA_ENDPOINT_PATH_COMPONENT_VALUE_TOKEN);
+            }
+            return errorResponse.entity("Invalid path to the oauth authorization server metadata document. " +
+                    "Received path : " + authzServerMetadataEpPathComponent + " is not resolvable").build();
+        }
+    }
+
+    private boolean isValidIssuer(String issuer) {
+
+        if (AUTHZ_SERVER_METADATA_ENDPOINT_PATH_COMPONENT_VALUE_TOKEN.equals(issuer)) {
+            return true;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("AuthzServerMetadataEndpointPathComponent validation failed. AuthzServerMetadataEndpointPathComponent value: " +
+                    issuer + ", not matched to '"
+                    + AUTHZ_SERVER_METADATA_ENDPOINT_PATH_COMPONENT_VALUE_TOKEN);
+        }
+        return false;
+    }
+
+    private Response getResponse(HttpServletRequest request, String tenant) {
+
+        String response;
+        OIDCProcessor processor = OIDCProviderServiceFactory.getOIDCService();
+        try {
+            AuthzServerMetadataJsonResponseBuilder responseBuilder = new AuthzServerMetadataJsonResponseBuilder();
+            response = responseBuilder.getAuthzServerMetadataConfigString(processor.getResponse(request, tenant));
+
+        } catch (OIDCDiscoveryEndPointException e) {
+            Response.ResponseBuilder errorResponse = Response.status(processor.handleError(e));
```

Comment:
In this case should we log server errors, if there's any

Reply:
This error is triggered, if there is a problem of fetching OIDC configurations in the BE, so I think in this case we should.